### ISSUE TITLE
IntuneDeviceAppManagement: New resource proposal

### ISSUE THREAD

**Issue by eclipse313** [microsoft/Microsoft365DSC#4585]
### Description
All resources of type [deviceAppManagement](https://learn.microsoft.com/de-de/graph/api/resources/intune-mam-deviceappmanagement?view=graph-rest-1.0) are missing. In particular, the option to manage mobile apps to ensure app installation on the end devices is missing. An example [Graph-API PowerShell](https://learn.microsoft.com/en-us/powershell/module/Microsoft.Graph.Devices.CorporateManagement/New-MgDeviceAppManagementMobileApp?view=graph-powershell-1.0) command is `New-MgDeviceAppManagementMobileApp`.

### Proposed properties
For me, the [mobile app](https://learn.microsoft.com/en-us/graph/api/resources/intune-apps-mobileapp?view=graph-rest-1.0) is currently the most important and the only thing I currently need. Note: Despite the name `mobileApp`, this can also be used to configure Windows apps. The implementation of the [assignments](https://learn.microsoft.com/en-us/graph/api/resources/intune-apps-mobileappassignment?view=graph-rest-1.0) is also necessary so that the apps can also be assigned.

### Special considerations or limitations
The Graph API is currently still in the beta version.

**Comment by eclipse313** [comment]
It looks like @NikCharlebois has already suggested implementing this in 2021. However, I can't find any further actions on this (see #1437).

**Comment by GeldHades27355** [comment]
Agreed. We deploy ~35 apps to our clients' devices (Windows, Android and iOS). This would allow us to automate so much more.

**Comment by ricmestre** [comment]
I still have #2996 opened for this.

**Comment by eclipse313** [comment]
Do you @NikCharlebois have an update for us on this? Or can someone else take care of it. Unfortunately I don't have the PowerShell skills to implement the feature in a push request myself.

**Comment by eclipse313** [comment]
Does anyone have an update? How can we contribute?
